File: cs101/pa4/Lex.c
//Andrew Nguyen
//anguy224
//Programming Assignment 2

#include<stdio.h>
#include<stdlib.h>
#include<string.h>

#include "List.h"

#define MAX_LEN 160

int main(int argc, char * argv[]){
   int count = 0;
   FILE *in, *out;
   char line[MAX_LEN];
   
   // check command line for correct number of arguments
   if( argc != 3 ){
      printf("Usage: %s <input file> <output file>\n", argv[0]);
      exit(1);
   }

   // open files for reading and writing 
   in = fopen(argv[1], "r");
   out = fopen(argv[2], "w");
   if( in==NULL ){
      printf("Unable to open file %s for reading\n", argv[1]);
      exit(1);
   }
   if( out==NULL ){
      printf("Unable to open file %s for writing\n", argv[2]);
      exit(1);
   }

   while (fgets(line, MAX_LEN, in) != NULL){
      count++;
   }
   
   rewind(in);   
   
   char lines[count - 1][MAX_LEN];
   int count2 = -1;
   
   while (fgets(line, MAX_LEN, in) != NULL){
      count2++;
      strcpy(lines[count2], line);
   }
   
   List list = newList();

   for (int i=0; i<count; i++) {
      char *tmp = lines[i];
      int j = i-1;
      moveBack(list);
      while (j>=0 && strcmp(tmp, lines[get(list)])<=0){
         j--;
         movePrev(list);
      }
      
      if(index(list) < 0)
         prepend(list, i);
      else
         insertAfter(list, i);
   }
   
   moveFront(list);
   while (index(list)>=0){
      fprintf(out, "%s", lines[get(list)]);
      moveNext(list);
   }

   freeList(&list);

   /* close files */
   fclose(in);
   fclose(out);
   

   return(0);
}

File: cs101/pa1/Lex.java
//Andrew Nguyen
//anguy224
//Program Assignment 1

import java.io.*;
import java.util.Scanner;

public class Lex
{
   public static void main(String[] args) throws IOException
   {
      Scanner input = null;
      PrintWriter output = null;
      String[] token = null;
      
      if (args.length!=2)
      {
         System.err.println("Usage: Lex infile outfile");
         System.exit(1);
      }
      
      input = new Scanner(new File(args[0]));
      
      int count = 0;
      while (input.hasNextLine())
      {
         count++;
         input.nextLine();
      }
      input.close();
      
      List newList = new List();
      token = new String[count];
      input = new Scanner(new File(args[0]));
      output = new PrintWriter(new FileWriter(args[1]));

      int count2 = -1;
      while (input.hasNextLine())
      {
         count2++;
         token[count2] = input.nextLine();
      }
      
      for (int i=0; i<token.length; i++)
      {
         String temp = token[i];
         
         int j = i-1;
         newList.moveBack();
         while (j>=0&&temp.compareTo(token[newList.get()])<=0)
         {
            j--;
            newList.movePrev();
         }
         
         if (newList.index() < 0){
            newList.prepend(i);
         }
         else
         {
            newList.insertAfter(i);
         }
      }
      
      newList.moveFront();
      while(newList.index()>=0) {
         output.println(token[newList.get()]);
         newList.moveNext();
      }
      
      input.close();
      output.close();
   }
}